### ISSUE TITLE
Modify constraints API to list foreign key constraints

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#943]
## Current behavior

- I can see unique and primary key constraints with a GET to `/tables/{table_id}/constraints/`, but the response does not include FK constraints.

## Desired behavior

- FK constraints should be included in the constraints response.
- Additional FK-specific details about each FK constraint should be present too, including:
    - Target columns of the FK
    - `ON DELETE` behavior
    - `ON UPDATE` behavior
    - `MATCH` behavior
    - `DEFERRED` behavior 

## Considerations
- A foreign key might be associated with more than one column, and in such cases will reference more than one column too. We're not planning to add support for creating multi-column FKs any time soon, but we do need support for displaying them, so the API needs to respond by showing all the constraint columns and all the target columns.
- Note that API consumers also need a way to associate each constraint column with its corresponding target column. If we return the constraint columns separately from the target columns, that's fine as long as the ordering of those two arrays is suitable for associating the columns by array index.
- The UI eventually want to display a specific warning in the front end when the user is deleting rows that will cascade in case of constraints with `ON DELETE CASCADE`.

## Spec Prompts
The person implementing this issue should spec out their proposed API changes and run them by the Mathesar core team. Questions the spec should answer include:
- How does the response indicate the target table and target columns?
- How does the response provide info about the FK-specific details listed above?

## Additional context
- Blocked by API changes in #898
- [Foreign Key Spec](https://wiki.mathesar.org/en/design/specs/foreign-key-constraints) -- Scenarios 2 and 3
- [Postgres docs for Constraints](https://www.postgresql.org/docs/13/ddl-constraints.html)
- [Postgres docs for `SET CONSTRAINTS`](https://www.postgresql.org/docs/13/sql-set-constraints.html)

**Comment by seancolsen** [comment]
@kgodey I'm assigning this to you to help answer the above questions and loop others in as needed.

**Comment by kgodey** [comment]
Updated description to handle open questions.
